pkg/vsphere/tasks/waiter.go

```diff
@@ -56,20 +57,21 @@ func Wait(ctx context.Context, f func(context.Context) (Task, error)) error {
 //       return vm, vm.Reconfigure(ctx, config)
 //    })
 func WaitForResult(ctx context.Context, f func(context.Context) (Task, error)) (*types.TaskInfo, error) {
-	var err error
 	var info *types.TaskInfo
 	var backoffFactor int64 = 1
 
 	for {
-		var t Task
-		if t, err = f(ctx); err == nil {
+		t, err := f(ctx)
+		log.Debugf("New task: %s", taskMoid(t))
+
+		if err == nil {
 			info, err = t.WaitForResult(ctx, nil)
```

Comment:
We didn't check to see if t was nil also. This panic's vic-machine when we reach this code.